src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,122 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private int size = 0;
+    private Object[] customArrayList = new Object[DEFAULT_SIZE];
```

Comment:
1 use generic type here
2 instantiate the array inside constructor
```suggestion
    private T[] customArrayList;
```